components/TextFields/src/MDCTextField.m

```diff
@@ -828,7 +828,7 @@ - (NSString *)accessibilityLabel {
   NSMutableArray *accessibilityStrings = [[NSMutableArray alloc] init];
   if ([super accessibilityLabel].length > 0) {
     [accessibilityStrings addObject:[super accessibilityLabel]];
-  } else if (self.placeholderLabel.accessibilityLabel.length > 0) {
+  } else if (!self.hidesPlaceholderOnInput && self.placeholderLabel.accessibilityLabel.length > 0) {
```

Comment:
Please add a unit test for this behavior if possible.  If it's not possible for some reason, please add testing steps to the commit message/PR summary.

Reply:
leaving bread crumbs: https://github.com/material-components/material-components-ios/pull/8830